clustermesh-apiserver deployment: support lifecycle and terminationGr…

acgs771126 opened the pull request:
clustermesh-apiserver deployment: support lifecycle and terminationGracePeriodSeconds

clustermesh-apiserver deployment support lifecycle and terminationGracePeriodSeconds, because we need a while time(by lifecycle.preStop hook) to wait AWS NLB draining target for prevent connection issue when clustermesh-apiserver pod terminating

Please note:  this change I hope can backported to 1.13 version

Fixes: #26900

```release-note
clustermesh-apiserver deployment support lifecycle and terminationGracePeriodSeconds.
```

kaworu commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -2801,6 +2804,12 @@ clustermesh:
+    # -- lifecycle setting for the apiserver container.
+    lifecycle: {}
+
+    # -- terminationGracePeriodSeconds(default 30s) for the clustermesh-apiserver deployment.
```
nit: no need to specify the default, the helm doc table has a column for the value and when reading the value file we can read the actual default value.
```suggestion
    # -- terminationGracePeriodSeconds for the clustermesh-apiserver deployment.
```

acgs771126 replied:
okay, let me update the PR, thanks.

kaworu reviewed (dismissed):
Thanks for the PR @acgs771126!

[CI failed](https://app.travis-ci.com/github/cilium/cilium/builds/264648114#L563) with:

> HINT: to fix this, run 'make -C Documentation update-helm-values'


Also please first report the changes from `install/kubernetes/cilium/values.yaml` to `install/kubernetes/cilium/values.yaml.tmpl` instead and then run:

```console
make -C install/kubernetes cilium/values.yaml
```

giorio94 commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -2721,6 +2721,9 @@ clustermesh:
       securityContext: {}
 
+      # -- lifecycle setting for the etcd container.
+      lifecycle: {}
```
Please add the the lifecycle configuration also to the kvstoremesh container below.

giorio94 reviewed (changes requested):
Thanks @acgs771126 for the PR!

One minor request inline on top of the comments from Alexandre.

giorio94 reviewed (changes requested):
Changes look good to me.

CI is failing with the following errors, which shall be fixed:

```
=========================================================
[1/1] Running on 6f0006779a9ecc757e06931cbe5facee053c0e43
clustermesh-apiserver deployment: support lifecycle and terminationGracePeriodSeconds
=========================================================
ERROR:NO_AUTHOR_SIGN_OFF: Missing Signed-off-by: line by nominal patch author 'andychuang <andy.chuang@shoplineapp.com>'

"[PATCH] clustermesh-apiserver deployment: support lifecycle and" has style problems, please review.

ERROR:CUSTOM: Please avoid long commit subjects (max: 75, found: 85)
```

Similarly, please also rerun `make -C install/kubernetes` and commit the changes (mending the current commit). 

(Edit) You should also run `Documentation/update-spelling_wordlist.sh terminationGracePeriodSeconds` and commit the changes.

zacharysarah commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -2718,6 +2718,9 @@ clustermesh:
       # -- Security context to be added to clustermesh-apiserver etcd containers
       securityContext: {}
 
+      # -- lifecycle setting for the etcd container.
```
No periods for sentence fragments
```suggestion
      # -- lifecycle setting for the etcd container
```

zacharysarah commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -2768,6 +2771,9 @@ clustermesh:
           drop:
             - ALL
 
+      # -- lifecycle setting for the KVStoreMesh container.
```
No periods for sentence fragments
```suggestion
      # -- lifecycle setting for the KVStoreMesh container
```

zacharysarah commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -2798,6 +2804,12 @@ clustermesh:
+    # -- lifecycle setting for the apiserver container.
+    lifecycle: {}
+
+    # -- terminationGracePeriodSeconds for the clustermesh-apiserver deployment.
```
No periods for sentence fragments
```suggestion
    # -- lifecycle setting for the apiserver container
    lifecycle: {}

    # -- terminationGracePeriodSeconds for the clustermesh-apiserver deployment
```

zacharysarah reviewed (changes requested):
@acgs771126 Some grammar nits, otherwise LGTM from a docs perspective.

Commit:
helm: apiserver support lifecycle and terminationGracePeriodSeconds

clustermesh-apiserver deploment support set lifecycle and terminationGracePeriodSeconds, because we need a while time(by lifecycle.preStop hook) to wait AWS NLB draining target for prevent connection issue when clustermesh-apiserver pod terminating

Fixes: #26900

Signed-off-by: andychuang <andy.chuang@shoplineapp.com>

zacharysarah reviewed (approved):
LGTM

tommyp1ckles commented on Documentation/spelling_wordlist.txt:
```diff
@@ -635,6 +635,7 @@ tcptop
 tcptracer
 templating
 terminatingTLS
+terminationGracePeriodSeconds
```
why do we need this, this isnt the first instance of this word

tommyp1ckles replied:
oh nevermind - missed the discussion regarding the commit message ci failure 😅

tommyp1ckles reviewed (approved):
looks good
